src/url-state-machine.js

```diff
@@ -93,6 +93,20 @@ function percentEncode(c) {
   return "%" + hex;
 }
 
+function percentDecode(input) {
```

Comment:
Can you comment on what was wrong with the old implementation? It seems mostly like things have been rearranged, and now all the callers have to toString() instead of it being done for them. I prefered the old implementation and name.

We certainly can't call it "percent decode", since that's defined in the spec as a byte sequence -> byte sequence operation. This used to be a string -> string operation, and now seems to be a string -> byte sequence operation.